runtime/misc.c

```diff
@@ -44,6 +44,9 @@ void caml_failed_assert (char * expr, char_os * file_os, int line)
           (Caml_state_opt != NULL) ? Caml_state_opt->id : -1, file, line, expr);
   fflush(stderr);
   caml_stat_free(file);
+#ifdef __GNUC__
```

Comment:
`__has_builtin(__builtin_trap) || __GNUC__` is probably better (it works with clang as well), see the definition of CAMLunreachable in misc.h (which also has a Windows version, by the way),

Reply:
Thanks, that's an improvement.

(For future reference: `__GNUC__` is defined by clang as well - it means "compiler supporting GCC extensions" rather than "GCC")